src/main/java/bio/terra/service/filedata/flight/ingest/IngestCopyLoadHistoryStep.java

```diff
@@ -50,48 +54,72 @@ public StepResult doStep(FlightContext context) {
     UUID datasetId = UUID.fromString(datasetIdString);
     Dataset dataset = datasetService.retrieve(datasetId);
 
-    int chunkNum = 0;
-    List<BulkLoadHistoryModel> loadHistoryArray = null;
     String flightId = context.getFlightId();
     String tableName_FlightId = flightId.replaceAll("[^a-zA-Z0-9]", "_");
     try {
       Instant loadTime = context.getStairway().getFlightState(flightId).getSubmitted();
-      bigQueryPdao.createStagingLoadHistoryTable(dataset, tableName_FlightId);
-      TimeUnit.SECONDS.sleep(waitSeconds);
+      LoadHistoryIterator historyIterator = loadService.loadHistoryIterator(loadId, fileChunkSize);
 
-      while (loadHistoryArray == null || loadHistoryArray.size() == fileChunkSize) {
-        loadHistoryArray =
-            loadService.makeLoadHistoryArray(loadId, fileChunkSize, (chunkNum * fileChunkSize));
-        chunkNum++;
-        // send list plus load_tag, load_time to BQ to be put in a staging table
-        if (!loadHistoryArray.isEmpty()) {
-          bigQueryPdao.loadHistoryToStagingTable(
-              dataset, tableName_FlightId, loadTag, loadTime, loadHistoryArray);
-          // Sleep to avoid BQ rate limit error
-          // From quick survey of logs, longest time to complete load query: 3 seconds
-          TimeUnit.SECONDS.sleep(waitSeconds);
-        }
+      if (platform.isGcp()) {
```

Comment:
Also, FWIW, i've been doing branching in the flight.  I'd suggest moving this logic into a separate Azure specific test

Reply:
Branching within the flight results in a lot of duplicated code which branching within the step avoids.

Reply:
I argue that this leads to a weird looking step where you're basically duplicating your steps.  We could have a common super class to encapsulate the reusable code but I like the idea of smaller less complicated steps (and consistency)